src/audio/module_adapter/module_adapter.c

```diff
@@ -677,6 +677,12 @@ int module_adapter_copy(struct comp_dev *dev)
 	}
 
 	if (mod->simple_copy) {
+		if (!source_c) {
+			ret = -ENODATA;
+			comp_err(dev, "comp_update_buffer_consume() %x: NULL dereference error",
+				 ret);
+			goto out;
+		}
```

Comment:
`source_c` can not be NULL, it is assigned at line 616 under the same if condition at line 612.

The static analyzer got this wrong by not understanding the context.

Reply:
And even if `source_c` is NULL (which it can not be), the `sink_c` might not be and still need to be released?

Reply:
Furthermore: at the `out:` label we have:
```c
if (mod->simple_copy) {
	buffer_release(sink_c);
	buffer_release(source_c);
}
```
where `source_c` is still NULL if we jump from here, is that fine?

Reply:
You're right. However in future there might be another check and it could pop ap again. My first approach was to add NULL prevent condition in comp_update_buffer_consume() because this function is causing that critical error. However exiting without dbg message was not good enough. What is your suggestion?

Reply:
Are we able to move on as a false positive today ?

Reply:
Ok, I'm closing it as false positive.